src/tests/queue-test/queue-test.c

```diff
@@ -127,28 +131,76 @@ void QueueTimeoutCheck(void)
                   (unsigned int)limit);
 }
 
-void UtTest_Setup(void)
+void QueueTimeoutSetup(void)
 {
-    if (OS_API_Init() != OS_SUCCESS)
+    int32  status;
+    uint32 accuracy;
+
+    task_1_failures = 0;
+    task_1_messages = 0;
+    task_1_timeouts = 0;
+
+    status = OS_QueueCreate(&msgq_id, "MsgQ", OSAL_BLOCKCOUNT_C(MSGQ_DEPTH), OSAL_SIZE_C(MSGQ_SIZE), 0);
+    UtAssert_True(status == OS_SUCCESS, "MsgQ create Id=%lx Rc=%d", OS_ObjectIdToInteger(msgq_id), (int)status);
+
+    /*
+    ** Create the "consumer" task.
+    */
+    status = OS_TaskCreate(&task_1_id, "Task 1", task_1, OSAL_STACKPTR_C(task_1_stack), sizeof(task_1_stack),
+                           OSAL_PRIORITY_C(TASK_1_PRIORITY), 0);
+    UtAssert_True(status == OS_SUCCESS, "Task 1 create Id=%lx Rc=%d", OS_ObjectIdToInteger(task_1_id), (int)status);
+
+    /*
+    ** Create a timer
+    */
+    status = OS_TimerCreate(&timer_id, "Timer 1", &accuracy, &(TimerFunction));
+    UtAssert_True(status == OS_SUCCESS, "Timer 1 create Id=%lx Rc=%d", OS_ObjectIdToInteger(timer_id), (int)status);
+    UtPrintf("Timer Accuracy = %u microseconds \n", (unsigned int)accuracy);
+
+    /*
+    ** Start the timer
+    */
+    status = OS_TimerSet(timer_id, timer_start, timer_interval);
+    UtAssert_True(status == OS_SUCCESS, "Timer 1 set Rc=%d", (int)status);
+
+    /* allow some time for task to run and accrue queue timeouts */
+    while (timer_counter < 100)
     {
-        UtAssert_Abort("OS_API_Init() failed");
+        OS_TaskDelay(100);
     }
+}
 
-    /*
-     * Register the test setup and check routines in UT assert
-     */
-    UtTest_Add(QueueTimeoutCheck, QueueTimeoutSetup, NULL, "QueueTimeoutTest");
+void QueueMessageCheck(void)
+{
+    int32 status;
+
+    OS_printf("Delay for half a second before checking\n");
+    OS_TaskDelay(500);
+
+    status = OS_TimerDelete(timer_id);
+    UtAssert_True(status == OS_SUCCESS, "Timer delete Rc=%d", (int)status);
+    status = OS_TaskDelete(task_1_id);
+    UtAssert_True(status == OS_SUCCESS, "Task 1 delete Rc=%d", (int)status);
+    status = OS_QueueDelete(msgq_id);
+    UtAssert_True(status == OS_SUCCESS, "Queue 1 delete Rc=%d", (int)status);
+
+    /* None of the tasks should have any failures in their own counters */
+    UtAssert_True(task_1_failures == 0, "Task 1 failures = %u", (unsigned int)task_1_failures);
+    UtAssert_True(task_1_messages == 10, "Task 1 messages = %u", (unsigned int)task_1_messages);
+    UtAssert_True(task_1_timeouts == 0, "Task 1 timeouts = %u", (unsigned int)task_1_timeouts);
 }
 
-void QueueTimeoutSetup(void)
+void QueueMessageSetup(void)
 {
     int32  status;
     uint32 accuracy;
-
     task_1_failures = 0;
     task_1_messages = 0;
     task_1_timeouts = 0;
 
+    int        i;
+    const char Data[4] = "xyz";
+
```

Comment:
nitpick - variable declarations should be grouped at beginning

Reply:
resolved